doc/design.md

```diff
@@ -0,0 +1,111 @@
+      Guest          +----------------------+
+      ============================================ <--- td-shim spec
+      Host           +----------------------+
+                     |         VMM          |
+                     +----------------------+
+
+   ```
+
+ The [td-shim specification](https://github.com/confidential-containers/td-shim/blob/main/doc/tdshim_spec.md) defines the interface between td-shim and vmm, and the interface between td-shim and td-payload.
+
+ The [td-shim threat model](https://github.com/confidential-containers/td-shim/blob/main/doc/threat_model.md) defines the threat model for the td-shim.
+
+This repo includes a full `td-shim`, and sample `td-payload`. The consumer may create other td-payload. For example, to support [TDX](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-trust-domain-extensions.html) [migration TD](https://www.intel.com/content/dam/develop/external/us/en/documents/tdx-migration-td-design-guide-348987-001.pdf), a rust-migtd can include a td-shim and a migtd-payload.
+
+## td-shim
+
+[rust-tdshim](https://github.com/confidential-containers/td-shim/tree/main/rust-tdshim) is a core of td-shim. The entrypoint is `_start()` at [main](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/src/main.rs). It will initialize the td-shim and switch to td-payload at `switch_stack_call()` of [main](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/src/main.rs).
+
+The TD_HOB is measured and event log is created at `create_td_event()` in [tcg.rs](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/src/tcg.rs).
+
+Data Execution Prevention (DEP) is setup at `find_and_report_entry_point()` in [ipl.rs](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/src/ipl.rs). The primitive `set_nx_bit()` and `set_write_protect()` are provided by [memory.rs](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/src/memory.rs).
+
+Control flow Enforcement Technology (CET) Shadow Stack is setup at `enable_cet_ss()` in [cet_ss.rs](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/src/cet_ss.rs).
+
+Stack guard is setup at `stack_guard_enable()` in [stack_guard.rs](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/src/stack_guard.rs).
+
+### reset vector
+
+[ResetVector](https://github.com/confidential-containers/td-shim/tree/main/rust-tdshim/ResetVector) is the reset vector inside of the td-shim. It owns the first instruction in TD at address 0xFFFFFFF0 - [resetVector](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/ResetVector/Ia32/ResetVectorVtf0.asm). Then it switches to long mode, parks APs, initializes the stack, copies the td-shim core to low memory (1MB) and call to rust-tdshim `call    rsi` at [main](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/ResetVector/Main.asm)
```

Comment:
The URLs are dangling until this is committed, but it's unclear why you are pointing in one case to the directory containing the various reset vector implementations, in the other to the IA32 one.

If I understood the intent correctly, what you mean is something like:

```
[ResetVector](https://github.com/confidential-containers/td-shim/tree/main/rust-tdshim/ResetVector) is the reset vector inside of the `td-shim`. It owns the first instruction in the TD at address 0xFFFFFFF0. This is implemented in the IA32 code named [resetVector](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/ResetVector/Ia32/ResetVectorVtf0.asm). The code then switches to long mode, parks application processors (APs), initializes the stack, copies the `td-shim` core to low memory (1MB) and call to `rust-tdshim` via an indirect call `call    rsi` at [main](https://github.com/confidential-containers/td-shim/blob/main/rust-tdshim/ResetVector/Main.asm)
```

(I am not entirely sure I captured what you meant).

Reply:
Yes, you are right.
I will use your sentence directly in next push. Thank you very much for the suggestion.